src/main/java/com/google/cloud/tools/app/deploy/DeployExample.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.deploy;
+
+import com.google.cloud.tools.app.deploy.gcloud.GcloudAppDeploy;
+import com.google.cloud.tools.app.deploy.process.LoggingOutputHandler;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.sdk.CloudSdk;
+import com.google.cloud.tools.app.impl.config.DefaultDeployConfiguration;
+
+import java.io.File;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+
+public class DeployExample {
+
+  public static void howToUse() {
+    // presumably the sdk object comes from somewhere else, but lets build it here
+    // you can also set enviroment on it
+    CloudSdk sdk = new CloudSdk.Builder().sdkPath(new File("some/path")).build();
+
+    // create the deploy object
+    Deploy deploy = GcloudAppDeploy.newBuilder(sdk).build();
```

Comment:
OutputHandler is the StatusMessageHandler you're looking for?

I mean the sdk potentially only needs to be configured and built once and can be passed around through an SdkManager/SdkService (or something). Since this implementation expects it to be a configuration only class.

```java
public class SdkManager/Service {
  CloudSdk instance
  setSdk()
  getSdk()
}
````

A global factory like this?
````java
class SdkBuilderFactory {
  void setSdkPath();
  void setEnvironment();
  Builder getBuilder() {
    return instance/new builder...
  }
````
 I think in the SDK case specifically, factory on the builder won't provide us much as I don't expect different SDKs to be built for the same application (unless the user starts changing path, perhaps though we trigger an update to the manager there).